plugins/auth-node/src/oauth/OAuthCookieManager.ts

```diff
@@ -152,6 +152,16 @@ export class OAuthCookieManager {
     };
     const req = res.req;
     let output = res;
+
+    // clear cookie with domain due to a bug during sign-in with redirect flow
+    // and invalid cookie being present that causes infinite redirect loop
+    const { hostname: domain } = new URL(this.options.callbackUrl);
+
+    output = output.cookie(name, '', {
+      ...this.getRemoveCookieOptions(),
+      domain,
+    });
```

Comment:
Hmm, this is a bit tricky since we can't really assume that we're not setting a domain since a custom cookie configurer might be installed. If there's a different solution that's fairly simple that might be better. Otherwise we might want to check what we're actually getting back from `.getRemoveCookieOptions()` and potentially skip this if a domain is set.

Reply:
As an update here -- I haven't been able to find a good way to address the infinite-looping issue on the frontend. The only approach that has worked is to set a query parameter after sign-in, to show that we've had a successful sign-in, and if we attempt to sign in again, we should stop the loop. This isn't great since the parameter shows up in the URL to the user.